014-quarkus-panache-with-transactions-xa/src/test/java/io/quarkus/qe/dbpool/AgroalPoolTest.java

```diff
@@ -0,0 +1,188 @@
+package io.quarkus.qe.dbpool;
+
+import static io.restassured.RestAssured.given;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.apache.http.HttpStatus;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+
+import io.agroal.api.AgroalDataSource;
+import io.quarkus.qe.containers.MysqlTestProfile;
+import io.quarkus.qe.rest.data.UserRepository;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.tuples.Tuple2;
+
+/**
+ * The aim of these tests is verified agroal and entityManager pool management
+ * Some of these tests required some extra load, in order to reproduce concurrency issues.
+ */
+@QuarkusTest
+@TestProfile(MysqlTestProfile.class)
+public class AgroalPoolTest {
+
+    private static final Logger LOG = Logger.getLogger(AgroalPoolTest.class);
+
+    @Inject
+    EntityManager em;
+
+    @Inject
+    UserRepository users;
+
+    @Inject
+    AgroalDataSource agroalDataSource;
+
+    static final int TIMEOUT_SEC = 60;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.idle-removal-interval")
+    String idleSec;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.max-size")
+    int datasourceMaxSize;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.min-size")
+    int datasourceMinSize;
+
+    @Test
+    @Disabled("Pending Zulip conversation about quarkus.datasource.jdbc.idle-removal-interval")
+    public void idleTimeoutTest() throws InterruptedException {
+        makeApplicationQuery();
+        Thread.sleep(Duration.ofSeconds(getIdleMs() + 1).toMillis());
```

Comment:
Is this working? You are parsing 1000 milliseconds + 1, then converting it to seconds, which is 1001 seconds and then to milliseconds 1001000. So, you're sleeping the test for 1001000 milliseconds.

Reply:
Was leftover as a result of a side effect from your previous comments.